flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasMultiClass.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.param;
+
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.StringParam;
+import org.apache.flink.ml.param.WithParams;
+
+/**
+ * Interface for the shared multi-class param.
+ *
+ * <p>Supported options:
+ * <li>auto: select the version based on the number of classes: If numClasses is one or two, set to
```

Comment:
For the long run, we could add `ClassificationModel ` and `Classifier `. But we should probably do it later since it would not introduce a lot of refatoring here.

Deciding `auto` has nothing to do with `ClassificationModel::numClasses` --- It is computed by the histogram of labels. Please feel free to checkout code at `LogisticRegression#L134` and Spark's implementation. [1]

[1] https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala#L519

Reply:
Thanks for the explanation. Sounds good.